sparse_ho/models/svm.py

```diff
@@ -23,49 +23,57 @@ class SVM(BaseModel):
     max_iter : int
         maximum number of epochs for the coordinate descent
         algorithm
-    tol : float
-        tolerance for the stopping criterion
+    estimator: instance of ``sklearn.base.BaseEstimator``
+        An estimator that follows the scikit-learn API.
     """
 
     def __init__(self, estimator=None, max_iter=100):
         self.estimator = estimator
         self.max_iter = max_iter
         self.dual = True
-        self.dr = None
+        self.dresiduals = None
 
-    def _init_dbeta_dr(self, X, y, dense0=None,
-                       mask0=None, jac0=None, compute_jac=True):
+    def _init_dbeta_dresiduals(
+            self, X, y, dense0=None, mask0=None, jac0=None, compute_jac=True):
         n_samples, n_features = X.shape
-        dr = np.ones(n_samples)
-        if self.dr is not None and self.dr.shape[0] == n_samples:
-            dr = self.dr.copy()
-        if issparse(X):
-            dbeta = (X.T).multiply(y * dr)
-            dbeta = np.sum(dbeta, axis=1)
-            dbeta = np.squeeze(np.array(dbeta))
+        dresiduals = np.zeros(n_samples)
+        if self.dresiduals is None:
+            dbeta = np.zeros(n_features)
         else:
-            dbeta = np.sum(y * dr * X.T, axis=1)
-        return dbeta, dr
-
-    def _init_beta_r(self, X, y, mask0, dense0):
+            if self.residuals.shape[0] != n_samples:
+                dbeta = np.zeros(n_features)
+            else:
+                dresiduals = self.dresiduals.copy()
+                if issparse(X):
+                    dbeta = (X.T).multiply(y * dresiduals)
+                    dbeta = np.sum(dbeta, axis=1)
+                    dbeta = np.squeeze(np.array(dbeta))
+                else:
+                    dbeta = np.sum(y * dresiduals * X.T, axis=1)
+        return dbeta, dresiduals
+
+    def _init_beta_residuals(self, X, y, mask0, dense0):
         n_samples, n_features = X.shape
-        r = np.zeros(n_samples)
-        if mask0 is None or self.r is None:
+        residuals = np.zeros(n_samples)
+        if mask0 is None or self.residuals is None:
             beta = np.zeros(n_features)
         else:
-            r = self.r
-            if issparse(X):
-                beta = (X.T).multiply(y * r)
-                beta = np.sum(beta, axis=1)
-                beta = np.squeeze(np.array(beta))
+            if self.residuals.shape[0] != n_samples:
+                beta = np.zeros(n_features)
             else:
-                beta = np.sum(y * r * X.T, axis=1)
-        return beta, r
+                residuals = self.residuals
+                if issparse(X):
+                    beta = (X.T).multiply(y * residuals)
+                    beta = np.sum(beta, axis=1)
+                    beta = np.squeeze(np.array(beta))
+                else:
+                    beta = np.sum(y * residuals * X.T, axis=1)
```

Comment:
```suggestion
                    beta = (y * residuals) * np.sum(X, axis=0)
```
if actually the same this would be much faster and you could precompute np.sum(X, axis=0)